### PR TITLE
docs: add examples to table expression API

### PR DESCRIPTION
**Pull request by cpcloud**
This PR fills out most of the table APIs examples sections with actual running examples.

### CODE REVIEW

File: ibis/expr/api.py [hunk 1]
```diff
@@ -453,18 +456,20 @@ def desc(expr: ir.Column | str) -> ir.Value:
...
-      metrics:
-        count: Count(t)
-      by:
-        g: r0.g
-    Selection[r1]
-      sort_keys:
-        desc|r1.count
+    >>> ibis.options.interactive = True
+    >>> t = ibis.examples.penguins.fetch()
+    >>> t[["species", "year"]].order_by(_.year.desc()).head()
```
Comment (gforsyth) [inline ibis/expr/api.py:461]: How is this completing if `_` hasn't been imported from `ibis`?

Comment (cpcloud) [reply ibis/expr/api.py:461]: Ah, because it's an interactive session, so `_` is `t` :)

That's not intentional, so I'll fix it up.

Nice catch.

Comment (cpcloud) [reply ibis/expr/api.py]: Fixed.

File: ibis/expr/types/relations.py [hunk 2]
```diff
@@ -263,62 +283,123 @@ def _ensure_expr(self, expr):
...
+        ┃ fruit  ┃ price   ┃
+        ┡━━━━━━━━╇━━━━━━━━━┩
+        │ string │ float64 │
+        ├────────┼─────────┤
+        │ apple  │    0.50 │
+        │ apple  │    0.50 │
+        │ banana │    0.25 │
+        │ orange │    0.33 │
+        └────────┴─────────┘
+        >>> t.group_by("fruit").agg(total_cost=_.price.sum(), avg_cost=_.price.mean())
```
Comment (gforsyth) [inline ibis/expr/types/relations.py:380]: Another errant `_`

File: ibis/expr/types/relations.py [hunk 3]
```diff
@@ -396,6 +477,32 @@ def aggregate(
...
+        ┃ fruit  ┃ price   ┃
+        ┡━━━━━━━━╇━━━━━━━━━┩
+        │ string │ float64 │
+        ├────────┼─────────┤
+        │ apple  │    0.50 │
+        │ apple  │    0.50 │
+        │ banana │    0.25 │
+        │ orange │    0.33 │
+        └────────┴─────────┘
+        >>> t.aggregate(by=["fruit"], total_cost=_.price.sum(), avg_cost=_.price.mean(), having=_.price.sum() < 0.5)
```
Comment (gforsyth) [inline ibis/expr/types/relations.py:497]: Needs `_` import

**Review by gforsyth — CHANGES_REQUESTED** [review]
This looks good!  Two more `_` to fix, but other than that I think this can go in.

File: ibis/expr/types/relations.py (hunk 2 above)
Comment (cpcloud) [reply ibis/expr/types/relations.py:380]: This all turns out to be because once you import something it stays in scope for the entire pytest session 😬

Comment (gforsyth) [reply ibis/expr/types/relations.py:609]: :grimacing: that is... not great

File: pyproject.toml [hunk 4]
```diff
@@ -281,22 +297,10 @@ filterwarnings = [
   "ignore:Deprecated call to `pkg_resources\\.declare_namespace\\('google.*'\\):DeprecationWarning",
   # pyspark on python 3.11
   "ignore:typing\\.io is deprecated:DeprecationWarning",
+  # warnings from google's use of the cgi module
```
Comment (gforsyth) [inline pyproject.toml:300]: ewww

**Review by gforsyth — APPROVED** [review]
this is great!  I think all the doctest examples are properly standalone.

**Pull request merged**